manager/src/main/java/com/redhat/service/bridge/manager/connectors/ConnectorsServiceImpl.java

```diff
@@ -49,30 +48,25 @@
         ConnectorEntity newConnectorEntity = new ConnectorEntity();
 
         newConnectorEntity.setName(newConnectorName);
-        newConnectorEntity.setStatus(ConnectorStatus.ACCEPTED);
-        newConnectorEntity.setDesiredStatus(ConnectorStatus.READY);
+        newConnectorEntity.setStatus(ManagedResourceStatus.ACCEPTED);
         newConnectorEntity.setSubmittedAt(ZonedDateTime.now());
         newConnectorEntity.setProcessor(processor);
         newConnectorEntity.setDefinition(connectorPayload);
         newConnectorEntity.setConnectorType(connectorType);
         newConnectorEntity.setTopicName(topicName);
 
         connectorsDAO.persist(newConnectorEntity);
-
-        return Optional.of(newConnectorEntity);
     }
 
     @Override
-    @Transactional
-    public List<ConnectorEntity> deleteConnectorIfNeeded(Processor processor) {
-
-        List<ConnectorEntity> optionalConnector = connectorsDAO.findByProcessorId(processor.getId());
-
-        return optionalConnector.stream().peek(c -> {
-            String connectorId = c.getId();
-            c.setDesiredStatus(ConnectorStatus.DELETED);
-            LOGGER.info("connector with id '{}' has been marked for deletion", connectorId);
-        }).collect(Collectors.toList());
+    @Transactional(Transactional.TxType.MANDATORY)
+    // Connector should always be marked for destruction in the same transaction as a Processor
+    public void deleteConnectorEntity(Processor processor) {
+        List<ConnectorEntity> connectors = connectorsDAO.findByProcessorId(processor.getId());
```

Comment:
@lucamolteni This is more a question for you: Why does this return a list? It's a one-to-one mapping between Processor and Connector isn't it?

Reply:
heheh we already discussed this in the past: it's a one-to-one mapped as a ManyToOne as one-to-one are tricky to model in hibernate

Reply:
I'll do on this PR.. There's been other related tidy-up so it won't hurt.

Reply:
Done.